server/routes/oodlet.js

```diff
@@ -70,33 +71,74 @@ let routes = [
   {
     method: 'GET',
     path: '/oodlet',
-    handler: list
+    config: {
+      handler: list,
+      validate: {
+        query: {
+          office: Joi.string().required(),
+          toDate: Joi.date(),
+          fromDate: Joi.date()
+        }
+      }
+    }
   },
   {
     method: 'GET',
     path: '/oodlet/{id}',
-    handler: get
+    config: {
+      handler: get,
+      validate: {
+        params: {
+          id: Joi.string().required()
+        }
+      }
+    }
   },
   {
     method: 'POST',
     path: '/oodlet',
-    handler: create
+    config: {
+      handler: create,
+      validate: {
+        payload: {
+          oodlerId: Joi.string().required(),
+          quantifiedThingies: Joi.array()
```

Comment:
Let's take this a step further. Let's create a custom `quantifiedThingy` JOI extension, and specify that this is not only an array, but an array of our new `quantifiedThingy` objects.